app/assets/javascripts/components/wizard/wizard.jsx

```diff
@@ -19,6 +19,24 @@ const getState = () =>
   })
 ;
 
+const persist = function () {
+  window.onbeforeunload = function () {
+      return "Data will be lost if you leave/refresh the page, are you sure?";
+  };
+  window.history.replaceState({ index: 0 }, 'wizard', '#step1'); // Initial States
+  document.title += ' — Step 1';
+  window.onpopstate = function (event) { // Listen to changes
+    if (event.state) {
+      WizardActions.goToWizard(event.state.index);
+    }
+  };
+};
+
+const unloadEvents = function () {
+    window.onpopstate = null;
+    window.onbeforeunload = null;
```

Comment:
These should probably be camelCase?

Reply:
@ragesoss They are vanilla JS events. They're not something which a user can define. We can only listen to these events. See [onpopstate](https://developer.mozilla.org/en-US/docs/Web/API/Window/onpopstate) & [onbeforeunload](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload)